src/cpp/session/modules/SessionFonts.cpp

```diff
@@ -0,0 +1,338 @@
+//    +-- Victor Mono Bold.woff
+//
+Error generateCssFromDir(const FilePath& dir,
+                         const std::string& fontName,
+                         const std::vector<std::string>& parents,
+                         std::string *pCss)
+{
+   std::vector<FilePath> files;
+   Error error = dir.getChildren(files);
+   if (error)
+   {
+      return error;
+   }
+   
+   for (const auto& file: files)
+   {
+      std::string name = file.getFilename();
+      if (file.isDirectory())
+      {
+         // This is a directory, probably representing a font weight or style
+         std::vector<std::string> newParents(parents);
+         newParents.push_back(name);
+         error = generateCssFromDir(file, fontName, newParents, pCss);
+      }
+      else if (isFontFile(file))
+      {
+         // This is an ordinary font file
+         error = generateCssFromFile(file, fontName, parents, pCss);
+      }
+
+      if (error)
+      {
+         // Log and clear any errors encountered while processing this font
+         LOG_ERROR(error);
+         error = Success();
+      }
+   }
+
+   return error;
+}
+
+// Handles an HTTP request for a specific font file.
+void handleFontFileRequest(const http::Request& request,
+                           http::Response* pResponse)
+{
+   std::string prefix = "/" kFontFiles;
+   std::string fileName = http::util::pathAfterPrefix(request, prefix);
+   
+   // Check user font folder first
+   FilePath fontFile = userFontFolder().completeChildPath(fileName);
+   if (!fontFile.exists())
+   {
+      // Not found in user fonts; fall back to system
+      fontFile = systemFontFolder().completeChildPath(fileName);
+   }
+
+   pResponse->setCacheableFile(fontFile, request);
+   return;
+}
+
+// Handles an HTTP request for font CSS. This will typically look something like:
+//
+// GET /fonts/css/Victor Mono.css
+//
+// The request is fulfilled by automatically generating the appropriate CSS @font-face rule(s) for
+// the font and returning them in the body of the request.
+void handleFontCssRequest(const http::Request& request,
+                          http::Response* pResponse)
+{
+   Error error;
+   std::string css;
+   std::string prefix = "/"  kFontCss;
+   std::string fileName = http::util::pathAfterPrefix(request, prefix);
+
+   // Strip off ".css" to get name of font
+   size_t idx = fileName.find(".css");
+   if (idx != std::string::npos)
+   {
+      fileName = fileName.substr(0, idx);
+   }
+   
+   // Enumerate user and system font directories
+   std::vector<FilePath> dirs;
+   dirs.push_back(userFontFolder());
+   dirs.push_back(systemFontFolder());
+
+   for (const auto& dir: dirs)
+   {
+      FilePath subDir = dir.completeChildPath(fileName);
+      if (subDir.exists() && subDir.isDirectory())
+      {
+         // Folder full of font files
+         std::vector<std::string> parents;
+         parents.push_back(fileName);
+         error = generateCssFromDir(subDir, fileName, parents, &css);
```

Comment:
Should you log this `error` or do something else with it, otherwise it's unused.